src/caw/caw_function.h

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2021, USC
+// All rights reserved.
+
+#ifndef SRC_CAW_CAW_FUNCTION_H_
+#define SRC_CAW_CAW_FUNCTION_H_
+
+#include <grpcpp/grpcpp.h>
+
+#include <string>
+#include <unordered_map>
+
+#include "../key_value/key_value_client.h"
+#include "caw.grpc.pb.h"
+
+namespace csci499 {
+
+using google::protobuf::Any;
+using grpc::Status;
+using grpc::StatusCode;
+
+using caw::RegisteruserRequest;
+using caw::RegisteruserReply;
+using caw::CawRequest;
+using caw::CawReply;
+using caw::FollowRequest;
+using caw::FollowReply;
+using caw::ReadRequest;
+using caw::ReadReply;
+using caw::ProfileRequest;
+using caw::ProfileReply;
+
+// struct for caw function reply message
+struct CawFuncReply {
```

Comment:
Why is this needed?  Can't you use the protobuf?

Reply:
I am passing both the reply status and serialized Reply message back through faz

Reply:
Hmm, I'm still a bit confused -- why isn't GRPC Status sufficient for the status?  You should be looking at the status (via GRPC) as separate from the reply message (which should be just the caw message, unless I'm missing something), but both come back from the server in the GRPC response.

Reply:
Caw, Read, and Profile all return serialized information, so I guess a better name for `CawFuncReply.message` would be `CawFuncReply.serialized_obj`.  GRPC Status allows you to return a status with a status message, but I did not want to store the serialized object in status message since I needed to pack it in an EventReply (protobuf Any) in Faz server.